numba/tests/test_tuples.py

```diff
@@ -484,5 +487,16 @@ def test_conversions(self):
                       str(raises.exception))
 
 
+class TestMethods(TestCase):
+
+    def test_index(self):
+        pyfunc = index_method_usecase
+        cfunc = jit(nopython=True)(pyfunc)
+        self.assertEqual(cfunc((1, 2, 3), 2), 1)
+
+        with self.assertRaises(ValueError):
```

Comment:
Please could you do something similar to https://github.com/numba/numba/pull/2807#discussion_r176710337 thanks.